main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -557,7 +557,7 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
     }
 
     def ForYield: Rule1[ParsedAst.Expression.ForYield] = rule {
-      SP ~ keyword("for") ~ optWS ~ ForFragments ~ optWS ~ keyword("yield") ~ WS ~ Expression ~ SP ~> ParsedAst.Expression.ForYield
+      SP ~ keyword("forM") ~ optWS ~ ForFragments ~ optWS ~ keyword("yield") ~ WS ~ Expression ~ SP ~> ParsedAst.Expression.ForYield
```

Comment:
Let us allow both `for` and `forM` for now-- but fine to refactor the test cases.

(Later `for` will become something that automatically selects between the two.)